Sources/Sentry/SentryTransportFactory.m

```diff
@@ -27,11 +27,19 @@ @implementation SentryTransportFactory
 + (NSArray<id<SentryTransport>> *)initTransports:(SentryOptions *)options
                                sentryFileManager:(SentryFileManager *)sentryFileManager
 {
-    NSURLSessionConfiguration *configuration =
-        [NSURLSessionConfiguration ephemeralSessionConfiguration];
-    NSURLSession *session = [NSURLSession sessionWithConfiguration:configuration
-                                                          delegate:options.urlSessionDelegate
-                                                     delegateQueue:nil];
+
+    NSURLSession *session;
+
+    if (options.urlSession) {
```

Comment:
`m`: Please add some tests to the `SentryTransportFactoryTests` for this new functionality.

Reply:
Updated unit test to cover 100% coverage for this file.